docs/using-modules.rst

```diff
@@ -0,0 +1,190 @@
+
+You can use this module's functionality simply by importing it, however any functionality that you do not use from a module will not be included in the final compilation target. For example, if you don't use the ``initializes`` statement to declare a module's location in the storage layout, you cannot use its state. Similarly, if you don't explicitly ``export`` an external function from a module, it will not appear in the runtime code.
+
+Importing a module
+==================
+
+A module can be imported using ``import`` or ``from ... import`` statements. The following are all equivalent ways to import the above module:
+
+.. code-block:: vyper
+
+    import ownable               # accessible as `ownable`
+    import ownable as ow         # accessible as `ow`
+    from . import ownable        # accessible as `ownable`
+    from . import ownable as ow  # accessible as `ow`
+
+When importing using the ``as`` keyword, the module will be referred to by its alias in the rest of the contract.
+
+The ``_times_two()`` helper function in the above module can be immediately used without any further work.
+
+.. code-block:: vyper
+
+    import ownable as helper
+    @external
+    def my_function(x: uint256) -> uint256:
+        return helper._times_two(x)
+
+The other functions cannot be used yet, because they touch the ``ownable`` module's state. There are two ways to declare a module so that its state can be used.
+
+Initializing a module
+=====================
+
+In order to use a module's state, it must be "initialized". A module can be initialized with the ``initializes`` keyword. This declares the module's location in the contract's :ref:`Storage Layout <compiler-storage-layout>`. It also creates a requirement to invoke the module's :ref:`__init__() function <init-function>`, if it has one. This is a well-formedness requirement, since it does not make sense to access a module's state unless its ``__init__()`` function has been called.
+
+.. code-block:: vyper
+
+    import ownable
+
+    initializes: ownable
+
+    @deploy
+    def __init__():
+        ownable.__init__()
+
+    @external
+    def my_access_controlled_function():
+        ownable._check_owner()  # reverts unless msg.sender == ownable.owner
+
+        ... # do things that only the owner can do
+
+It is a compile-time error to invoke a module's ``__init__()`` function more than once!
+
+A module's state can be directly accessed just by prefixing the name of a variable with the module's alias, like follows:
+
+.. code-block:: vyper
+
+    @external
+    def get_owner() -> address:
+        return ownable.owner
+
```

Comment:
Add a note that a stateless modules must not be initialised.

Reply:
we should make it an error tbh

Reply:
sure, sgtm

Reply:
https://github.com/vyperlang/vyper/issues/4050